plutus-tx/src/PlutusTx/Foldable.hs

```diff
@@ -43,13 +43,18 @@ import           Data.Monoid           (First (..))
 import           Data.Semigroup        (Dual (..), Endo (..), Product (..), Sum (..))
 import           GHC.Exts              (build)
 import           PlutusTx.Applicative  (Applicative (pure), (*>))
-import           PlutusTx.Bool         (not)
+import           PlutusTx.Base
+import           PlutusTx.Bool         (Bool (..), not)
+import           PlutusTx.Builtins     (Integer)
+import           PlutusTx.Either       (Either (..))
 import           PlutusTx.Eq           (Eq (..))
-import           PlutusTx.Functor      (id)
+import           PlutusTx.Maybe        (Maybe (..))
 import           PlutusTx.Monoid       (Monoid (..))
 import           PlutusTx.Numeric      (AdditiveMonoid, AdditiveSemigroup ((+)), MultiplicativeMonoid)
 import           PlutusTx.Semigroup    ((<>))
-import           Prelude               (Bool (..), Either (..), Integer, Maybe (..), Monad (..), flip, (.))
+
+-- TODO: define Plutus Monad?
```

Comment:
Well, the fact that we don't have it tells us that all the functions in here that rely on it are broken, and we should probably delete them!

Reply:
Hmm. Turned out we also export `return` in `Prelude` because sometimes we use do notation for lists. I guess then we can afford these monad functions as well. And also I remember that we added `mapM_` by user request.